packages/smooth_app/lib/pages/product/add_basic_details_page.dart

```diff
@@ -125,30 +130,63 @@ class _AddBasicDetailsPageState extends State<AddBasicDetailsPage> {
                     if (!_formKey.currentState!.validate()) {
                       return;
                     }
-                    final bool savedAndRefreshed =
-                        await ProductRefresher().saveAndRefresh(
-                      context: context,
-                      localDatabase: localDatabase,
-                      product: Product(
-                        productName: _productNameController.text,
-                        quantity: _weightController.text,
-                        brands: _brandNameController.text,
-                        barcode: _product.barcode,
+                    final Product inputProduct = Product(
+                      productName: _productNameController.text,
+                      quantity: _weightController.text,
+                      brands: _brandNameController.text,
+                      barcode: _product.barcode,
+                    );
+                    final String uniqueId =
```

Comment:
You're generating ids in multiple places, please do in it in a single one

Reply:
Since everytime i require a unique id with those parameters so I am creating at places where i need them

Reply:
Yes, but a Id generator or something like would be better to understand.
Your code is mixed along many things